dataset/src/test/scala/frameless/functions/AggregateFunctionsTests.scala

```diff
@@ -93,4 +131,114 @@ class AggregateFunctionsTests extends TypedDatasetSuite {
     check(forAll(prop[Int] _))
     check(forAll(prop[Byte] _))
   }
+
+  test("max") {
+    def prop[A: TypedEncoder : Ordering](xs: List[A])(
+      implicit
+      ex1: TypedEncoder[X1[A]],
+      eoa: TypedEncoder[Option[A]]
+    ): Prop = {
+      val dataset = TypedDataset.create(xs.map(X1(_)))
+      val A = dataset.col[A]('a)
+      val datasetMax::_ = dataset.select(max(A)).collect().run().toList
```

Comment:
Maybe you could `.head` instead? I'm not really sure why Spark returns List[Option[A]] here...